doc/programming/parsing.rst

```diff
@@ -359,9 +359,19 @@ currently being parsed:
     ``self``.
 
 ``$$``
-    Inside field attributes and hooks, ``$$`` refers to the just
-    parsed value, even if it's not going to be directly stored in the
-    field. The value of ``$$`` is writable and may be modified.
+    Inside field attributes, ``$$`` refers to the value as it was
+    parsed. Inside field hooks, ``$$`` refers to the final value
+    *after* any conversions are applied (see
+    :ref:`attribute_convert`). This applies even if the value is not
+    going to be directly stored in the field. The value of ``$$`` is
+    writable and may be modified.
+
+.. note::
+
+   ``$$`` has slightly different semantics in a field attribute and
+   in a hook. In an attribute, ``$$`` refers to the parsed value
+   *before* any conversions. In a hook, ``$$`` refers to the final
+   value *after* any conversions.
```

Comment:
This note is slightly redundant with the explanation in `$$`, but I think it's worth it to hammer the point home. It also introduces the difference in terms from "parsed" and "final" value.